core/common/src/main/java/alluxio/resource/DynamicResourcePool.java

```diff
@@ -336,15 +323,15 @@ public T acquire(long time, TimeUnit unit)
    */
   @Override
   public void release(T resource) {
+    if (!mResources.containsKey(resource)) {
+      throw new IllegalArgumentException(
+          "Resource " + resource.toString() + " was not acquired from this resource pool.");
+    }
+    ResourceInternal<T> resourceInternal = mResources.get(resource);
+    resourceInternal.setLastAccessTimeMs(mClock.millis());
     try {
       mLock.lock();
-      if (!mResources.containsKey(resource)) {
-        throw new IllegalArgumentException(
-            "Resource " + resource.toString() + " was not acquired from this resource pool.");
-      }
-      ResourceInternal<T> resourceInternal = mResources.get(resource);
-      resourceInternal.setLastAccessTimeMs(mClock.millis());
-      mResourceAvailable.add(resourceInternal);
+      mResourceAvailable.addFirst(resourceInternal);
```

Comment:
I don't think we need to do that because i think it is user's responsibility that release/acquire must be paired. It is similar as how semaphore works. Our old ResourcePool works this way.